### DISCUSSION TITLE
Can i train models in keras 3 and run them in keras 2?

### DISCUSSION THREAD

**Discussion by satyrmipt** [keras-team/keras#20227] [Q&A] [answered]
I used to train models in google colab and they migrate to keras 3.1.4 from keras 2.15.
Production (inference) system is 2.15 and customer is not ready to update it.
Is there any info about "reverse" back compatibility when you train models in new keras and run them on older keras?

Model example: keras.applications.EfficientNetV2L
Custom_objects: yes, i use custom loss function

**Comment by abhaskumarsinha** [comment, marked as answer]
Hello @satyrmipt 

Any keras v3.0 model cannot run on keras v2+. The new `keras.ops.*` operations might need to be reverted to `tf.*` operations. The issues related to optimizer incompatibility might appear. Training of the model/model loop, if using custom loops is totally different in two versions and for the best results, the latest version has major bugfixes. It is highly recommended to stick to the latest version, along with porting the code to the new version according to the guide: https://keras.io/guides/migrating_to_keras_3/

Keras v2.* may not receive any major updates in the future and lacks the latest features including - quantization, LoRA, etc that are essential for modern models.

Best Regards,
Abhas Kumar Sinha
